GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -310,67 +345,172 @@ - (instancetype)init {
                            selector:@selector(macOSApplicationWillTerminate:)
                                name:NSApplicationWillTerminateNotification
                              object:nil];
-#endif  // TARGET_OS_IOS || TARGET_OS_TV
+
+#elif TARGET_OS_WATCH
+    // TODO: Notification on watchOS platform is currently posted by strings which are frangible.
+    // TODO: Needs improvements here.
+    NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
+    [notificationCenter addObserver:self
+                           selector:@selector(iOSApplicationDidEnterBackground:)
+                               name:@"UIApplicationDidEnterBackgroundNotification"
+                             object:nil];
+    [notificationCenter addObserver:self
+                           selector:@selector(iOSApplicationWillEnterForeground:)
+                               name:@"UIApplicationWillEnterForegroundNotification"
+                             object:nil];
+
+    // Adds observers for app extension on watchOS platform
+    [notificationCenter addObserver:self
+                           selector:@selector(iOSApplicationDidEnterBackground:)
+                               name:NSExtensionHostDidEnterBackgroundNotification
+                             object:nil];
+    [notificationCenter addObserver:self
+                           selector:@selector(iOSApplicationWillEnterForeground:)
+                               name:NSExtensionHostWillEnterForegroundNotification
+                             object:nil];
+#endif
   }
   return self;
 }
 
+/** Generates and maps a unique background identifier to the given semaphore.
+ *
+ * @param semaphore The semaphore to map.
+ * @return A unique GDTCORBackgroundIdentifier mapped to the given semaphore.
+ */
++ (GDTCORBackgroundIdentifier)createAndMapBackgroundIdentifierToSemaphore:
```

Comment:
Should the next two functions be wrapped in `#if TARGET_OS_WATCH`

Reply:
Same as above.

Reply:
LGTM